packages/framework/src/schemas/steps/channels/in-app.schema.ts

```diff
@@ -1,10 +1,20 @@
 import { Schema } from '../../../types/schema.types';
 
+const redirectSchema = {
+  type: 'object',
+  properties: {
+    url: { type: 'string' },
```

Comment:
We should also add the `format` field wherever possible to narrow the schema down.

```suggestion
    url: { type: 'string', format: 'uri' },
```